### PR TITLE
DIV-4929 Apply fix for document mapping and checking if date is empty string

### PR DESCRIPTION
**Pull request by alexwakeman**
We have a bug in Prod where we see Parse errors when trying to parse an empty string.
See ticket for more details. 

[Prevent Parse errors when document date is empty string](https://tools.hmcts.net/jira/browse/DIV-4929)

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/divorce/caseformatterservice/mapper/DocumentCollectionDivorceFormatMapper.java [hunk 1]
```diff
@@ -23,19 +32,32 @@
...
             .map(Document::getDocumentLink)
             .map(DocumentLink::getDocumentUrl)
             .ifPresent(url -> documentUrlRewrite.getDocumentId(url).ifPresent(result::setId));
     }
+
+    @AfterMapping
+    protected void mapCreatedDateCheckEmpty(CollectionMember<Document> document,
+                                            @MappingTarget  UploadedFile result) {
+        try {
+            if (document.getValue().getDocumentDateAdded() != null && !document.getValue().getDocumentDateAdded().isEmpty()) {
```
Comment (qzhou-hmcts) [inline src/main/java/uk/gov/hmcts/reform/divorce/caseformatterservice/mapper/DocumentCollectionDivorceFormatMapper.java]: CollectionUtils.isEmpty()

File: src/main/java/uk/gov/hmcts/reform/divorce/caseformatterservice/mapper/DocumentCollectionDivorceFormatMapper.java [hunk 2]
```diff
@@ -23,19 +32,32 @@
...
+                SimpleDateFormat dateFormat = new SimpleDateFormat(SIMPLE_DATE_FORMAT);
+                Date date = new Date(dateFormat.parse(document.getValue().getDocumentDateAdded()).getTime());
+                result.setCreatedOn(date);
+            }
+        } catch (ParseException ex) {
+            log.warn("Unable to parse a date string on document");
```
Comment (qzhou-hmcts) [inline src/main/java/uk/gov/hmcts/reform/divorce/caseformatterservice/mapper/DocumentCollectionDivorceFormatMapper.java]: Don't swallow this exception, if is the wrong format better propagate it

File: src/main/java/uk/gov/hmcts/reform/divorce/caseformatterservice/mapper/DocumentCollectionDivorceFormatMapper.java [hunk 3]
```diff
@@ -23,19 +32,32 @@
     private DocumentUrlRewrite documentUrlRewrite;
 
     @Mapping(source = "value.documentLink.documentFilename", target = "fileName")
-    @Mapping(source = "value.documentDateAdded", dateFormat = SIMPLE_DATE_FORMAT, target = "createdOn")
```
Comment (qzhou-hmcts) [inline src/main/java/uk/gov/hmcts/reform/divorce/caseformatterservice/mapper/DocumentCollectionDivorceFormatMapper.java]: do we know why this is empty?

Comment (alexwakeman) [reply src/main/java/uk/gov/hmcts/reform/divorce/caseformatterservice/mapper/DocumentCollectionDivorceFormatMapper.java]: No I am still investigating why we are getting an empty date value back.

File: src/main/java/uk/gov/hmcts/reform/divorce/caseformatterservice/mapper/DocumentCollectionDivorceFormatMapper.java (hunk 2 above)
Comment (alexwakeman) [reply src/main/java/uk/gov/hmcts/reform/divorce/caseformatterservice/mapper/DocumentCollectionDivorceFormatMapper.java]: Would rather catch and then log it, so we can debug if required. Given the parser may throw this, it requires a try / catch

File: src/main/java/uk/gov/hmcts/reform/divorce/caseformatterservice/mapper/DocumentCollectionDivorceFormatMapper.java (hunk 1 above)
Comment (alexwakeman) [reply src/main/java/uk/gov/hmcts/reform/divorce/caseformatterservice/mapper/DocumentCollectionDivorceFormatMapper.java]: Not sure what this is for?

**Comment by nsingamsetty** [comment]
This is the PR to fix the Exceptions in Azure prod.

**Pull request merged**